### PR TITLE
avoid rhn re-registration on every ursula run.

### PR DESCRIPTION
**Pull request by nirajdp76**
This PR is to address below rhn subscription optimizations:
  - Register system only when it is not subscribed to rhn
  - dislablae/enabled rhel repos only for new subscription or refresh repo is true

### CODE REVIEW

File: roles/rhn-subscription/tasks/main.yml [hunk 1]
```diff
@@ -7,11 +12,18 @@
...
+    with_items: "{{ rhn_subscription.repos.disable }}"
 
-- name: enable desire redhat repositories
-  command: subscription-manager repos --enable={{ rhn_subscription.repos.enable | join(' --enable=') }}
+  - name: enable desire redhat repositories
+    command: subscription-manager repos --enable={{ rhn_subscription.repos.enable | join(' --enable=') }}
+    register: result
+    until: result|succeeded
+    retries: 3
+  when: subscription.changed or rhn_subscription.repos.refresh|bool
```
Comment (chengtcli) [inline roles/rhn-subscription/tasks/main.yml]: we don't need `bool` filter here.
I made some tests with ansible 2.1.0.0, 2.1.3.0. ansible can recognize true/false,True/False,'true'/'false','True'/'False', and regard them as bool

**Comment by nirajdp76** [comment]
ok to test

**Comment by nirajdp76** [comment]
ok to test

File: roles/rhn-subscription/tasks/main.yml (hunk 1 above)
Comment (paulczar) [inline roles/rhn-subscription/tasks/main.yml]: We've hit too many edge cases where something doesn't treat these as bool correctly that its a safeguard worth having.

**Comment by nirajdp76** [comment]
ok to test

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
ok to test
